presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+            if (this.getViewNames(viewName.getSchemaName()).contains(viewName.getTableName())) {
+                throw new PrestoException(ALREADY_EXISTS, "View already exists");
+            }
+
+            if (this.getTableNames(viewName.getSchemaName()).contains(viewName.getTableName())) {
+                throw new PrestoException(ALREADY_EXISTS, "View already exists as data table");
+            }
+        }
+
+        if (replace && getView(viewName) != null) {
+            metaManager.deleteViewMetadata(viewName);
+        }
+
+        metaManager.createViewMetadata(new AccumuloView(viewName.getSchemaName(), viewName.getTableName(), viewData));
+    }
+
+    /**
+     * Drops the given view
+     *
+     * @param viewName View to drop
+     */
+    public void dropView(SchemaTableName viewName)
+    {
+        metaManager.deleteViewMetadata(viewName);
+    }
+
+    /**
+     * Adds a new column to an existing table
+     *
+     * @param table Accumulo table to act on
+     * @param column New column to add
+     * @throws IndexOutOfBoundsException If the ordinal is greater than or equal to the number of columns
+     */
+    public void addColumn(AccumuloTable table, AccumuloColumnHandle column)
+            throws IndexOutOfBoundsException
+    {
+        // Add the column
+        table.addColumn(column);
+
+        // Recreate the table metadata with the new name and exit
+        metaManager.deleteTableMetadata(new SchemaTableName(table.getSchema(), table.getTable()));
+        metaManager.createTableMetadata(table);
+
+        // Validate index tables exist -- User may have added an index column to a previously
+        // un-indexed table
+        if (table.isIndexed() && !table.isExternal()) {
+            if (!conn.tableOperations().exists(table.getIndexTableName())) {
+                createIndexTables(table);
+            }
+        }
+    }
+
+    /**
+     * Rename the column of an existing table
+     *
+     * @param table Accumulo table to act on
+     * @param source Existing column name
+     * @param target New column name
+     */
+    public void renameColumn(AccumuloTable table, String source, String target)
+    {
+        if (table.getRowId().equals(source)) {
+            table.setRowId(target);
+        }
+
+        boolean found = false;
+        // Locate the column to rename
+        for (AccumuloColumnHandle col : table.getColumns()) {
+            if (col.getName().equals(source)) {
+                found = true;
+
+                // Rename the column
+                col.setName(target);
+
+                // Recreate the table metadata with the new name and exit
+                metaManager.deleteTableMetadata(
+                        new SchemaTableName(table.getSchema(), table.getTable()));
+                metaManager.createTableMetadata(table);
+                break;
+            }
+        }
+
+        if (!found) {
```

Comment:
Instead of this `if` simply `return` in loop (when found) and here unconditionally throw.